### ISSUE TITLE
Chat transfers private message height to new window

### ISSUE THREAD

**Issue by vipperz** [drawpile/Drawpile#1375]
I don't remember it doing this before, but if you would still have a private message tab in a session window and open another window from the same one (using New/ Open.../ Open recent/ Join session), the chat will stay the same size as if the private message is still there.

I'm not sure when this started happening in the Continuous Release, but it's gotten more noticeable since I keep the chat at minimal height.

**Comment by askmeaboutlo0m** [comment]
The chat remembers its height and if you open a new window, it will continue to have that height. That doesn't have anything to do with the tabs inside of the chat.

Is the issue that the tabs make the chat higher? They do need a bit of extra space, so they expand the chat height if necessary.

**Comment by vipperz** [comment]
Yes, it makes it higher, but the size of docks and chat wouldn't remember height/width until a Drawpile window was closed and re-opening a new window right after.

Having to close the private message for the chat to stay the original size in a new window wasn't needed until updating the Continuous Release, at some point.

**Comment by askmeaboutlo0m** [comment]
If it didn't remember the height before, that was a bug. It's supposed to write the main window configuration when opening a new one.

That the chat changes size with tabs appearing and disappearing is mildly annoying though, so making that work differently could be implemented and would solve the issue.

**Comment by vipperz** [comment]
Oh, if it were a bug, it's what I've gotten used to over the years without much thought. Having the chat unchangeable in a way would be so nice, though.

Side note: If someone where to have two or more windows open - changing brush settings, adding new colors to a palette group, etc. in one - and doesn't close the changed window last, the changes get lost or not applied.

I'm not sure if this is considered a bug too or it's implied to happen, because it can be frustrating when you realize you didn't close that specific session last.

**Comment by askmeaboutlo0m** [comment]
It's definitely undesirable behavior, but a pretty hard problem. Synchronizing *everything* between the windows leads to other undesirable behavior, like docks changing position, brushes being changed from under you etc. So it would really need a complete overhaul of the settings to figure out what should be synchronized and what should be per-window.

**Comment by MorrowShore** [comment]
Will the chat system redesign not solve these issues on its own?

**Comment by askmeaboutlo0m** [comment]
Depends on how chat tabs are implemented for it. If they use a regular Qt tab bar, it springing into existence would change the minimum height of the widget it's part of.
